ssl/ssl_rsa.c

```diff
@@ -57,6 +57,11 @@ int SSL_use_certificate_file(SSL *ssl, const char *file, int type)
     int ret = 0;
     X509 *cert = NULL, *x = NULL;
 
+    if (file == NULL) {
+        ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_NULL_PARAMETER);
+        goto file_end;
```

Comment:
Hmm, as a one-off I was not going to comment, but as a pattern used repeatedly:

We already have `end` as a label for an error exit. Rather than adding another label, in the generally "happy" path, why not initialise `in` to `NULL` in the parameter list above and `goto end` here? Adding `file_end` promotes this unlikely "file parameter was NULL" input condition to something more important than it needs to be

Reply:
Sure, I can reuse `end` for file. I did not think about that before. Sorry about that.